templates/_macros/quote-wrapper.jinja

```diff
@@ -0,0 +1,138 @@
+{#
+  Params
+  - heading_text (string) (optional): The text to be displayed as the heading
+  - quote_heading_level (int) (optional): The heading level used for the quote text. Can be 2 or 4. Defaults to 2.
+  - quote_text (string) (required): The text of the quote. The macro will surround it with quotes, so there is no need to quote it yourself.
+  - citation_name (string) (optional): The name of the source being quoted
+  - citation_title (string) (optional): The title of the source being quoted
+  Slots
+  - signpost_image (optional): The signpost_image of the source being quoted
+  - heading_link (optional): Link to be displayed beside the heading text
+  - cta (optional): Contents of the call to action block to be displayed below the quote
+  - image (optional): An image to be displayed below the quote
+#}
+{%- macro quote_wrapper(
+  heading_text="",
+  quote_heading_level=2,
```

Comment:
I'm hesitant to abstract this that far, in case design at some point wants us to support quote with text size that is not `<h2>` or `<h4>`.

Reply:
I would also consider abstracting it. Especially that we already seem to limit that just to "2, 4, 6", so it's not like we allow any possible heading level.

It seems to be a quote size option. The fact that it matches to one of heading sizes is a bit of implementation detail.

I guess again, the question is: how is this documented and presented from design perspective - in figma. Is this marked as a heading level in Figma, or as a size of "Quote pattern" in Figma? Because in the end, that's what matters: developers will be looking into Figma, and need to translate this into props of a pattern.

Reply:
These were presented as heading levels in Figma. It is not difficult to translate them into a `quote_size` param with options of `large`, `medium`, `small`. However, that would make it difficult for this to support any other heading level sizes later.

Reply:
Abstracted this into `large`/`medium`/`small`, design does not think we will need more sizes ([ref](https://chat.canonical.com/canonical/pl/9qi85kurijr59n9yqu7dni7y5h))